tests/unit/bounding_box/test_bounding_box.py

```diff
@@ -0,0 +1,197 @@
+    docstring = get_docstring_from_node(None)
+    assert docstring == "No result found."
+
+
+def test_fetch_bounding_box_with_class_node():
+    node = parse("class Foo:\n    pass\n")
+    class_node = node.body[0]
+    bounding_box = fetch_bounding_box(class_node)
+    assert bounding_box.top_left.line == 1
+    assert bounding_box.top_left.column == 0
+    assert bounding_box.bottom_right.line == 2
+    assert bounding_box.bottom_right.column == 8
+
+
+def test_get_node_without_docstrings():
+    node = parse("def foo():\n" '    """This is a docstring."""\n' "    pass\n")
+    node_without_docstrings = get_node_without_docstrings(node.body[0])
+    assert get_docstring_from_node(node_without_docstrings) == "No result found."
+
+
+def test_get_node_without_imports():
+    node = parse("import os\ndef foo():\n" '    """This is a docstring."""\n' "    pass\n")
+    module_node = cast(ast.Module, node)
+    node_without_imports = get_node_without_imports(module_node)
+    assert isinstance(node_without_imports.body[0], ast.FunctionDef)  # skip the import statement
+
+
+@pytest.mark.parametrize(
+    "source,expected",
+    [
+        ('def foo():\n    """This is a docstring."""\n    pass\n', "This is a docstring."),
+        (
+            'class Foo:\n    """This is a class docstring."""\n    pass\n',
+            "This is a class docstring.",
+        ),
+        ("foo = 10", "No result found."),
+        ("import os", "No result found."),
+        ("from os import path", "No result found."),
+        ("# this is a comment", "No result found."),
+        (
+            '"""This is a standalone docstring."""\n"""Another standalone string."""',
+            "This is a standalone docstring.",
+        ),
+    ],
+)
+def test_get_docstring_from_node_with_various_nodes(source, expected):
+    module = parse(source)
+    if module.body and isinstance(
+        module.body[0], (ast.AsyncFunctionDef, ast.ClassDef, ast.FunctionDef, ast.Module)
+    ):
+        node = module.body[0]
+    else:
+        node = module
+    docstring = get_docstring_from_node(node)
+    assert docstring == expected
+
+
+@pytest.mark.parametrize(
+    "source",
+    [
+        'def foo():\n    """This is a docstring."""\n    pass\n',
+        'class Foo:\n    """This is a class docstring."""\n    pass\n',
+        "foo = 10",
+        '"""This is a standalone docstring."""',
+    ],
+)
+def test_get_node_without_docstrings_with_various_nodes(source):
+    node = parse(source)
+    node_without_docstrings = get_node_without_docstrings(node)
+    assert get_docstring_from_node(node_without_docstrings) == "No result found."
+
+
+@pytest.mark.parametrize(
+    "source",
+    [
+        "import os\ndef foo():\n    pass\n",
+        "from os import path\ndef foo():\n    pass\n",
+        "def foo():\n    pass\n",
+    ],
+)
+def test_get_node_without_imports_with_various_nodes(source):
+    node = parse(source)
+    node_without_imports = get_node_without_imports(node)
```

Comment:
Are you testing the bounding box here? I think test_get_node and the like should probably live in another test file, like test_py_ast_utils.py. Then instead of 'bounding_box' these tests could live in 'py_ast'

Reply:
Changed structure to better encapsulate this with the SRP as
tests/
```
│
└───unit/
    │
    └───ast_utils/
        │
        ├── __init__.py
        ├── test_ast_utils_bounding_box.py
        └── test_ast_utils.py
```